### PR TITLE
LPS-200611 Search bar can't find objects on folders

### PR DESCRIPTION
**Pull request by Artartur**
Related Issue: [LPS-200611](https://liferay.atlassian.net/browse/LPS-200611)

### CODE REVIEW

**Comment by Artartur** [comment]
ci:test:sf

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/LeftSidebar/LeftSidebar.tsx [hunk 1]
```diff
@@ -64,7 +64,11 @@ export default function LeftSidebar({setShowModal}: LeftSidebarProps) {
 			(key) => key === selectedObjectFolder.name
 		) as string;
 
-		setExpandedKeys(new Set([selectedObjectFolderKey]));
+		const filteredFolders = leftSidebarItems
+			.filter((item) => item.leftSidebarObjectDefinitionItems?.length)
+			.map((filteredNames) => filteredNames.name);
```
Comment (cgmonte) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/LeftSidebar/LeftSidebar.tsx:69]: `filteredNames` is not an array of names, but and array of items, returned by the filter array method. I would name this `filteredItem`, in the singular.

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/LeftSidebar/LeftSidebar.tsx [hunk 2]
```diff
@@ -64,7 +64,11 @@ export default function LeftSidebar({setShowModal}: LeftSidebarProps) {
...
+
+		setExpandedKeys(new Set([selectedObjectFolderKey, ...filteredFolders]));
```
Comment (cgmonte) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/LeftSidebar/LeftSidebar.tsx:71]: This solution is making everything expanded at first render. The correct behavior is to have the selected folder expanded and rest collapsed, at first render. The right behavior is to have the selected folder expanded and the rest collapsed at first render. Expansion should initially controlled by the search matches when there is a search key entered.

Also, there is something wrong with `selectedObjectFolderKey`. If uncategorized is not selected, `selectedObjectFolderKey` is `undefined`, which adds `undefined` as the first item of `expandedKeys` array. This does  not look right and may be related to our problem.

**Review by cgmonte — CHANGES_REQUESTED** [review]
Hi @Artartur, thank you very much for sending this. Although it fixes the bug, this solution introduces an unexpected behavior, and also the application's state data related to the panel expansions is inconsistent. Please, read my comments and let me know if it makes sense. I think we will need to rework this.

**Review by alinedoleron — APPROVED** [review]
LGTM! I tested and it's good

**Comment by alinedoleron** [comment]
ci:test:relevant

**Comment by alinedoleron** [comment]
ci:test:object

**Comment by carolmariaabb** [comment]
ci:forward:force

**Comment by carolmariaabb** [comment]
ci:forward:force

**Pull request closed without merging**